### PR TITLE
Add mfc app; Add volume filter

### PR DESCRIPTION
**Pull request by vrajendra-datacore**
With MFC app, we can select the particular drive or all drives (except network drives) to monitor the write requests.
Added necessary code in driver to achieve same.

MFC app

### CODE REVIEW

File: FsFilter1/FsFilter1.cpp [hunk 1]
```diff
@@ -94,31 +108,138 @@ FLT_PREOP_CALLBACK_STATUS MiniPreWrite(PFLT_CALLBACK_DATA Data, PCFLT_RELATED_OB
...
+	return STATUS_SUCCESS;
+}
+
 NTSTATUS MiniUnload(FLT_FILTER_UNLOAD_FLAGS Flags)
 {
 	UNREFERENCED_PARAMETER(Flags);
 	
-	KdPrint(("Driver Unload.\r\n"));
+	TraceEvent(TRACE_INFO, "Driver Unload.\r\n");
+	FltCloseCommunicationPort(port);
```
Comment (datacore-gsaini) [inline FsFilter1/FsFilter1.cpp]: Can we call this on already closed port?

Comment (vrajendra-datacore) [reply FsFilter1/FsFilter1.cpp]: Yes, may be we need NULL check for port before we Close the port.
And if it is closed, assign NULL to it.

File: FsFilter1/FsFilter1.cpp [hunk 2]
```diff
@@ -74,6 +84,10 @@ FLT_PREOP_CALLBACK_STATUS MiniPreWrite(PFLT_CALLBACK_DATA Data, PCFLT_RELATED_OB
 		status = FltParseFileNameInformation(FileNameInfo);
 		if (NT_SUCCESS(status)) {
 			RtlCopyMemory(FileName, FileNameInfo->Name.Buffer, FileNameInfo->Name.MaximumLength);
+			if (wcsncmp(FileName, drive_filter.Buffer, wcslen(drive_filter.Buffer)) != 0) {
+				KdPrint(("%ws is filtered out. Does not match %ws. Returning back\n", FileName, drive_filter.Buffer));
```
Comment (datacore-gsaini) [inline FsFilter1/FsFilter1.cpp:93]: Should we call FltReleaseFileNameInformation in this block before returning?

Comment (datacore-gsaini) [reply FsFilter1/FsFilter1.cpp:93]: And also I think we should add the same logic to MiniPreCreate also.

**Comment by datacore-gsaini** [comment]
Rest of the changes look fine to me.

File: FsFilter1/FsFilter1.cpp (hunk 2 above)
Comment (vrajendra-datacore) [reply FsFilter1/FsFilter1.cpp:93]: Addressed

**Pull request merged**